apps/pipeline/test/integration/pipeline/writer/table_writer_test.exs

```diff
@@ -142,67 +157,79 @@ defmodule Pipeline.Writer.TableWriterTest do
 
       datum = TDG.create_data(dataset_id: dataset.id, payload: payload)
 
-      expected = [
-        "Joe",
-        10,
-        ["bob", "sally"],
-        [["Bill", "Bunco"], ["Sally", "Bosco"]],
-        ["Susan", "female", ["Joel", "1941-07-12"]]
-      ]
+      expected = %{
+        "age" => "10",
+        "first_name" => "Joe",
+        "friend_names" => ["bob", "sally"],
+        "friends" => [%{"first_name" => "Bill", "pet" => "Bunco"}, %{"first_name" => "Sally", "pet" => "Bosco"}],
+        "spouse" => %{
+          "first_name" => "Susan",
+          "gender" => "female",
+          "next_of_kin" => %{"date_of_birth" => "1941-07-12", "first_name" => "Joel"}
+        }
+      }
 
       assert :ok = TableWriter.write([datum], table: dataset.technical.systemName, schema: schema)
 
       eventually(fn ->
+        query = "select * from foo__baz"
+
         result =
-          "select * from foo__baz"
-          |> Prestige.execute()
-          |> Prestige.prefetch()
+          session
+          |> Prestige.execute!(query)
+          |> Prestige.Result.as_maps()
 
         assert result == [expected]
       end)
     end
   end
 
   describe "compact/1" do
-    test "compacts a table without changing data" do
+    test "compacts a table without changing data", %{session: session} do
       sub = [%{name: "three", type: "boolean"}]
-      schema = [%{name: "one", type: "list", itemType: "integer"}, %{name: "two", type: "map", subSchema: sub}]
+      schema = [%{name: "one", type: "list", itemType: "decimal"}, %{name: "two", type: "map", subSchema: sub}]
```

Comment:
why are we changing these types? this type has proven to be ... problematic in the past. ~Jeff